src/XML/src/FMI3/fmi3_xml_variable.c

```diff
@@ -1165,7 +1165,8 @@ static int fmi3_xml_variable_process_attr_causality_variability_initial(fmi3_xml
     }
     if (!fmi3_is_valid_variability_causality(variability, causality)) {
         fmi3_variability_enu_t bad_variability = variability;
-        variability = fmi3_get_default_valid_variability(causality);
+        bool isFloat = (elm_id == fmi3_xml_elmID_Float32Variable) || (elm_id == fmi3_xml_elmID_Float64Variable);
```

Comment:
Could you use `variable->baseType` instead? I'm trying to reduce places where elmID is used for anything but choosing the element handler + error reporting.

Like this:

`isFloat = (fmi3_base_type_enu_is_float(fmi3_xml_get_variable_base_type(variable)))`

Reply:
Using `fmi3_xml_get_variable_base_type` was also my first choice. 
Unfortunately, `variable->type` is only set later on. Not overly happy with this solution, but anything else would require a larger restructuring at this point.